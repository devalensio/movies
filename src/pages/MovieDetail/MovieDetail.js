import React, { useEffect } from 'react';

import { connect } from 'react-redux';

import styles from './MovieDetail.module.scss';

const MovieDetail = ({ dispatch, history, match, movieDetail }) => {
  useEffect(() => {
    dispatch({
      type: 'GET_MOVIE_DETAIL',
      payload: { movieId: match.params.id },
    })

    return () => dispatch({
      type: 'RESET_MOVIE_DETAIL',
      payload: {},
    });
  }, []);

  return (
    <div className={styles.movie__detail}>
      <div className={styles['movie__detail--header']}>
        <div className={styles.header__icon} onClick={() => history.goBack()}>
          <img src="/images/back-arrow.png" alt="search-icon" width="16px" height="16px" />
        </div>
        <h2>Movie Detail</h2>
      </div>
      <div className={styles['movie__detail--body']}>
        <img src={movieDetail.Poster === 'N/A' ? '/images/not-found.png' : movieDetail.Poster} alt="Movie Poster" />
        <h1>{movieDetail.Title}</h1>
      </div>
      <p>Type: {movieDetail.Type || '-'}</p>
      <p>Release Date: {movieDetail.Year || '-'}</p>
      <p>Genre: {movieDetail.Genre || '-'}</p>
      <p>Language: {movieDetail.Language || '-'}</p>
      <p>Origin: {movieDetail.Country || '-'}</p>
      <p>Story: {movieDetail.Plot || '-'}</p>
    </div>
  );
};

const mapStateToProps = (state) => state.movies;

export default connect(mapStateToProps)(MovieDetail);
