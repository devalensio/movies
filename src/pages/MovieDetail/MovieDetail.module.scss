@import "../../styles/tokens/all";

.movie__detail {
  padding: $space--md;
  @media (min-width: 768px) {
    @include surface("2", null);
    border-radius: $space--2xs;
  }
  &--header {
    display: flex;
    align-items: center;
    margin-bottom: $space--md;
    > h2 {
      margin: 0;
      margin-left: $space--md;
    }
    .header__icon {
      cursor: pointer;
    }
  }
  &--body {
    display: flex;
    flex-direction: column;
    h1 {
      text-align: center;
    }
  }
}
