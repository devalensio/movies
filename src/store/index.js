import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';

import rootReducers from './shell/rootReducers';
import rootSagas from './shell/rootSagas';

const sagaMiddleware = createSagaMiddleware();

export const store = createStore(
  rootReducers,
  applyMiddleware(sagaMiddleware),
);

sagaMiddleware.run(rootSagas);

export default store;
