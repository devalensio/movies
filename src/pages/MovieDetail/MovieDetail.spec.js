import React from 'react';

import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import MovieDetail from './MovieDetail';
import store from '../../store';


describe('Movie Detail', () => {
  it('renders MovieDetail successfully', () => {
    const component = renderer.create(
      <Provider store={store}>
        <MovieDetail />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
