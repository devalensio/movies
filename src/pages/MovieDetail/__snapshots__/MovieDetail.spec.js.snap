// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Detail renders MovieDetail successfully 1`] = `
<div
  className="movie__detail"
>
  <div
    className="movie__detail--header"
  >
    <div
      className="header__icon"
      onClick={[Function]}
    >
      <img
        alt="search-icon"
        height="16px"
        src="/images/back-arrow.png"
        width="16px"
      />
    </div>
    <h2>
      Movie Detail
    </h2>
  </div>
  <div
    className="movie__detail--body"
  >
    <img
      alt="Movie Poster"
    />
    <h1 />
  </div>
  <p>
    Type: 
    -
  </p>
  <p>
    Release Date: 
    -
  </p>
  <p>
    Genre: 
    -
  </p>
  <p>
    Language: 
    -
  </p>
  <p>
    Origin: 
    -
  </p>
  <p>
    Story: 
    -
  </p>
</div>
`;
